src/cholesky/lkjcholesky.jl

```diff
@@ -0,0 +1,183 @@
+struct LKJCholesky{T <: Real, D <: Integer} <: Distribution{CholeskyVariate,Continuous}
+    d::D
+    η::T
+    uplo::Char
+    logc0::T
+end
+
+#  -----------------------------------------------------------------------------
+#  Constructors
+#  -----------------------------------------------------------------------------
+
+function LKJCholesky(d::Integer, η::Real, _uplo::Union{Char,Symbol} = 'L'; check_args = true)
+    if check_args
+        d > 0 || throw(ArgumentError("Matrix dimension must be positive."))
+        η > 0 || throw(ArgumentError("Shape parameter must be positive."))
+    end
+    logc0 = lkj_logc0(d, η)
+    uplo = _as_char(_uplo)
+    uplo ∈ ('U', 'L') || throw(ArgumentError("uplo must be 'U' or 'L'."))
+    T = Base.promote_eltype(η, logc0)
+    LKJCholesky{T, typeof(d)}(d, T(η), uplo, T(logc0))
+end
+
+_as_char(c::Char) = c
+_as_char(x) = only(string(x))
+
+#  -----------------------------------------------------------------------------
+#  REPL display
+#  -----------------------------------------------------------------------------
+
+show(io::IO, d::LKJCholesky) = show_multline(io, d, [(:d, d.d), (:η, d.η), (:uplo, d.uplo)])
+
+#  -----------------------------------------------------------------------------
+#  Conversion
+#  -----------------------------------------------------------------------------
+
+function convert(::Type{LKJCholesky{T}}, d::LKJCholesky) where T <: Real
+    return LKJCholesky{T, typeof(d.d)}(d.d, T(d.η), d.uplo, T(d.logc0))
+end
+function convert(::Type{LKJCholesky{T}}, d::Integer, η, logc0) where T <: Real
+    return LKJCholesky{T, typeof(d)}(d, T(η), d.uplo, T(logc0))
+end
+
+#  -----------------------------------------------------------------------------
+#  Properties
+#  -----------------------------------------------------------------------------
+
+dim(d::LKJCholesky) = d.d
+
+function size(d::LKJCholesky)
+    p = dim(d)
+    return (p, p)
+end
+
+function insupport(d::LKJCholesky, R::Cholesky)
+    p = dim(d)
+    factors = R.factors
+    (isreal(factors) && size(factors, 1) == p) || return false
+    iinds, jinds = axes(factors)
+    T = typeof(one(eltype(R))^2)
+    # check that the diagonal of U'*U or L*L' is all ones
+    @inbounds if R.uplo === 'U'
+        for j in 1:p
+            s = zero(T)
+            for i in 1:j
+                s += factors[iinds[i], jinds[j]]^2
+            end
+            isapprox(s, 1) || return false
+        end
+    else  # R.uplo === 'L'
+        for i in 1:p
+            s = zero(T)
+            for j in 1:i
+                s += factors[iinds[i], jinds[j]]^2
+            end
+            isapprox(s, 1) || return false
+        end
+    end
+    return true
+end
+
+function mode(d::LKJCholesky)
+    p = dim(d)
+    factors = Matrix{partype(d)}(I, p, p)
```

Comment:
In general, the type of the parameters is not equivalent with the (element-)type of the samples/support.